src/trans/CMakeLists.txt

```diff
@@ -32,41 +32,207 @@ if(CMAKE_Fortran_COMPILER_ID MATCHES "Intel")
+      list(APPEND outfiles ${outfile})
+      string(TOUPPER ${outfile_name_we} OUTFILE_NAME_WE )
+      ecbuild_debug("Generate ${destination}/trans_${backend}/${outfile_name}")
+      file(WRITE  ${destination}/trans_${backend}/${outfile_name} "! Automatically generated interface header for backward compatibility of generic symbols !\n")
+      file(APPEND ${destination}/trans_${backend}/${outfile_name} "#if defined(${outfile_name_we})\n")
+      file(APPEND ${destination}/trans_${backend}/${outfile_name} "#undef ${outfile_name_we}\n")
+      file(APPEND ${destination}/trans_${backend}/${outfile_name} "#endif\n")
+      file(APPEND ${destination}/trans_${backend}/${outfile_name} "#if defined(${OUTFILE_NAME_WE})\n")
+      file(APPEND ${destination}/trans_${backend}/${outfile_name} "#undef ${OUTFILE_NAME_WE}\n")
+      file(APPEND ${destination}/trans_${backend}/${outfile_name} "#endif\n")
+      file(APPEND ${destination}/trans_${backend}/${outfile_name} "#include \"${outfile_name_we}_${backend}${outfile_ext}\"\n")
+      file(APPEND ${destination}/trans_${backend}/${outfile_name} "#define ${outfile_name_we} ${OUTFILE_NAME_WE}_${backend}\n")
+      file(APPEND ${destination}/trans_${backend}/${outfile_name} "#define ${OUTFILE_NAME_WE} ${OUTFILE_NAME_WE}_${backend}\n")
+    endif()
+  endforeach(file_i)
+
+  add_custom_target(${_PAR_TARGET}_generate DEPENDS ${outfiles})
+  ecbuild_add_library(TARGET ${_PAR_TARGET} TYPE INTERFACE)
+  add_dependencies(${_PAR_TARGET} ${_PAR_TARGET}_generate)
+  target_include_directories(${_PAR_TARGET} INTERFACE $<BUILD_INTERFACE:${destination}>)
+endfunction(generate_backend_includes)
+
+
+
+function(generate_backend_sources)
+  set (options)
+  set (oneValueArgs BACKEND DESTINATION OUTPUT)
+  set (multiValueArgs)
+
+  cmake_parse_arguments(_PAR "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+  set(backend ${_PAR_BACKEND})
+  set(destination ${_PAR_DESTINATION})
+  file(MAKE_DIRECTORY ${destination}/algor)
+  file(MAKE_DIRECTORY ${destination}/internal)
+  file(MAKE_DIRECTORY ${destination}/external)
+
+  ecbuild_list_add_pattern( LIST files
+    GLOB
+          algor/*
+          internal/*
+          external/*
+    QUIET
+  )
+  if( NOT HAVE_FFTW )
+    ecbuild_list_exclude_pattern( LIST files REGEX tpm_fftw.F90 )
+  endif()
+
+
+  set(outfiles)
+  foreach(file_i ${files})
+    if(NOT (${file_i} IN_LIST ectrans_common_src))
+      get_filename_component(outfile_name    ${file_i} NAME)
+      get_filename_component(outfile_name_we ${file_i} NAME_WE)
+      get_filename_component(outfile_ext     ${file_i} EXT)
+      get_filename_component(outfile_dir     ${file_i} DIRECTORY)
+      set(outfile "${destination}/${file_i}")
+      ecbuild_debug("Generate ${outfile}")
+      generate_file(BACKEND ${backend} INPUT ${file_i} OUTPUT ${outfile})
+      list(APPEND outfiles ${outfile})
+    endif()
+  endforeach(file_i)
+  set(${_PAR_OUTPUT} ${outfiles} PARENT_SCOPE)
```

Comment:
I wasn't aware of `return(PROPAGATE)`. It is very neat but I see it is only available since cmake 3.25 which is too recent to be a requirement. This is a known idiom. Perhaps in the distant future this may be cleaned up in the entire stack, if it's worth it as it is only syntactic.

Reply:
Sure, I didn't notice that is 3.25 up.